src/domain/states/objects/fileTree/SchemaTree.ts

```diff
@@ -34,6 +34,10 @@ export class SchemaTree {
         this.forEachDirRecursive(this._children, fn);
     }
 
+    public forEachFile(fn: (file: SchemaFile) => void): void {
```

Comment:
Here my usual concerns apply:

- Will this recursion scale long term?
- If executed inside of a rendering call, it maybe would make sense to wrap calls to this function with a `useMemo` hook or remove it out of the rendering function as this is a rather expensive operation

I guess this is the JS or TS way to have a function like this taking a callback as a param. But IMHO I prefer to pull the functionality of the callbacks down into the class and make them available via the class interface.

This has the advantage that no outside caller can abuse the callback mechanism 😅 
We would just expose the valid operations and nothing more 😄

But please discuss these points, when necessary with the BE devs.

Reply:
I am using the same approach as in forEachDir() (L. 33) and getFile() (L. 29) here and since the recursion limit is somewhere far above 1000x, this will never be a problem as long as the tool is not used in a project with a file that is deeper than 1000+ folders.
Also the operation is not as expensive as it might seem, as during testing I was not able to notice any type of lag or performance issues at all during any actions.
Since the project deadline is tomorrow I would vote to keep it as is for now and talk about this topic again afterwards to discuss possible optimisations here if need be, especially since this would then also would have to be changed for forEachDir() and getFile() as well.